README.md

```diff
@@ -1,19 +1,149 @@
 # Kashmir [![NPM version][npm-image]][npm-url]
-> Generator for the Kashmir Drupal theme
+> A Drupal theme generator
 
 ## Installation
+---
+
+Install generator-kashmir using [npm](https://www.npmjs.com/) (we assume you have pre-installed [node.js](https://nodejs.org/)).
 
-First, install [Yeoman](http://yeoman.io) and generator-kashmir using [npm](https://www.npmjs.com/) (we assume you have pre-installed [node.js](https://nodejs.org/)).
 
 ```bash
-npm install -g yo generator-kashmir
+npm install -g generator-kashmir
 ```
 
-Then generate your new theme:
+Then generate your new theme using the generator. Use the below command from the directory where you want the theme to be generated.
 
 ```bash
-yo kashmir
+kashmir
 ```
 
 [npm-image]: https://badge.fury.io/js/generator-kashmir.svg
 [npm-url]: https://npmjs.org/package/generator-kashmir
+
+## Design systems
+---
+
+The theme provides two options for design systems, [Storybook.js](https://storybook.js.org/) and [Patternlab](https://patternlab.io/). These are included as part of the theme and you can choose one of the two during the theme generation process.
+
+The recommended option going forward would be to use Storybook as it is more modern and supports component building for numerous JS frameworks.
+
+### Storybook.js
+
+Refer [Storybook.js](https://storybook.js.org/) documentation for additional information about stories.
+
+
+### Compile and launch storybook
+
+```bash
+yarn storybook
+```
+
+### Patternlab
+
+```<<<< Placeholder >>>>```
+
+## Component generation
+---
+
+Once theme is generated you can use the theme generator to build components for the theme.
+
+Below is a example of generating story for *button* component,
+
+```bash
+╰─❯ kashmir
+? Please choose what you would like to generate? component
+? Please enter the component name: button
+? Choose the type of the component atom
+? Choose the files you to generate? sass, json, twig, markdown, stories
+The button component was created.
+```
+The button component is already included as an example and can be viewed as part of the storybook out-of-the-box.
+
+### Component structure
+
+The theme genertor would ask you which files you want to generate for the the component,
+- sass - *will contain the styles for component*
+- json - *will contain the data for the component, ex: title, description.*
+- twig - *will contain the markup for the component*
+- markdown - *documentation*
+- stories - *the story js file*
+- typescript - *additional js file*
+
+Based on your component needs, choose which files you need. 
+
+Please check the button component example provided with the theme.
+
+## Build commands
+---
+
+### Compile theme
+
+In order to compile the assets of the theme, this also compiles sass inside components,
+
+```bash
+yarn gulp
```

Comment:
@swarad07 Running `gulp` will suffice here.

And also this step needs to come before running the storybook or pattern lab as it will require the global CSS changes otherwise those commands will throw errors. Let's move this section before the Design Systems section as anyways this needs to happen before in case folks don't want to have a design system or even when they have a design system they would need to run this before running design system-specific build commands.

Reply:
I have made this change in https://github.com/swarad07/kashmir/pull/1 for review.